src/Stripe.Tests.XUnit/_infrastructure/encoding_and_decoding_enums.cs

```diff
@@ -0,0 +1,95 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using FluentAssertions;
+using Xunit;
+using Stripe.Infrastructure;
+
+namespace Stripe.Tests.Xunit
+{
+    public class encoding_and_decoding_enums
+    {
+        [JsonConverter(typeof(StringEnumConverter))]
+        public enum TestEnum
+        {
+            [EnumMember(Value = "test_one")]
+            TestOne,
+
+            // TestTwo doesn't define a serialization value
+            TestTwo,
+        }
+
+        public class TestObject
+        {
+            [JsonProperty("enum")]
+            public TestEnum? Enum { get; set; }
+        }
+
+        public class TestService : StripeService
+        {
+            public TestService() : base(null)
+            {
+            }
+        }
+
+        public encoding_and_decoding_enums()
+        {
+        }
+
+        [Fact]
+        public void encoding_should_use_enummember_value_if_it_exists()
+        {
+            var obj = new TestObject
+            {
+                Enum = TestEnum.TestOne,
+            };
+            var url = new TestService().ApplyAllParameters(obj, "", false);
+            url.Should().Be("?enum=test_one");
+        }
+
+        [Fact]
+        public void encoding_should_convert_to_string_if_no_enummember_value()
+        {
+            var obj = new TestObject
+            {
+                Enum = TestEnum.TestTwo,
+            };
+            var url = new TestService().ApplyAllParameters(obj, "", false);
+            url.Should().Be("?enum=TestTwo");
```

Comment:
Should we throw an error here? It seems that if we forgot an `EnumMember`, it was probably an accident that we should try to find out about before it starts causing problems.

Reply:
That makes sense, but it's not possible with the current implementation as I'm actually delegating the encoding to `JsonConverter` ([here](https://github.com/stripe/stripe-dotnet/blob/3f78d8bf6ca49b3c9fb5c5ca71e91c57b9844a38/src/Stripe.net/Infrastructure/Middleware/ParserPlugin/EnumPlugin.cs#L20)) which will transparently fall back to using the constant name as a string if no `EnumMember` is present.

It would probably be better to use reflection to fetch the `EnumMember`'s `Value` ourselves, and that's what I tried to do initially but I couldn't manage it with my (basically inexistent) .NET skills.

Happy to revisit this, or delegate to someone with experience in dealing with reflection in .NET.

Reply:
Cool. Alright, I don't think it's that big of a deal. Thanks for the explanation.

Reply:
What you have works, but I think [this is what you are looking for](https://gist.github.com/jaymedavis/b1d773064d860dec0408fffe2ed30a5b)?